### DISCUSSION TITLE
hx --health rust is perfect, "Language server exited"

### DISCUSSION THREAD

**Discussion by bq-wrongway** [helix-editor/helix#7671] [Troubleshooting]
I deleted whole helix, and rust, and installed everything again from scratch, and no luck.
The new install i did i build helix from source, and linked up everything, setting also HELIX_RUNTIME:
```
 hx --health rust 
Configured language server: rust-analyzer
Binary for language server: /home/melnibone/.cargo/bin/rust-analyzer
Configured debug adapter: lldb-vscode
Binary for debug adapter: /home/melnibone/devtools/llvm/bin/lldb-vscode
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```
For example zig works with no issues, the only issue is rust. The moment i enter any .rs file, i get 
"Language server exited" error , i am at my wits end. 
i am on pop os, i am using fish as my default and login shell, llvm-config version reports 16.4 (i dont know if this matters).

I had helix working with 0 issues on different machine.

**Comment by gabydd** [comment]
The log file https://github.com/helix-editor/helix/wiki/faq#access-the-log-file should have some more information that would be helpful for us to help you debug this issue

**Comment by bq-wrongway** [comment]
I got this :
```
3023-07-18T18:30:27.156 helix_lsp::transport [ERROR] rust-analyzer err <- "error: 'rust-analyzer' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'\n"
2023-07-18T18:30:27.157 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-07-18T18:30:27.157 helix_lsp::transport [ERROR] rust-analyzer err: <- StreamClosed
```

When i saw this i thought that for some reason rust analyzer is not being picked up, so i took binary of rust analyzer from their site and just add it to the path, and now it works.

But now i have two rust_analyzers:
~/.local/bin
~/.cargo/bin
Both of them are in $PATH, i am not sure why is this happening, or if i perhaps caused this somehow, this was a clean system, and i installed helix like i do every time. Is it safe to delete one of the rust analyzers ?

**Reply by the-mikedavis** [reply]
The `rust-analyzer` that was on your path looks like a wrapper script from rustup and it looks like you just need to install rust-analyzer for the stable toolchain. https://rust-analyzer.github.io/manual.html#rustup
